cloud_info_provider/core.py

```diff
@@ -217,37 +71,32 @@ def parse_opts():
     parser.add_argument(
         '--middleware',
         metavar='MIDDLEWARE',
-        choices=SUPPORTED_MIDDLEWARE,
+        choices=providers,
         default='static',
         help=('Middleware used. Only the following middlewares are '
               'supported: %s. If you do not specify anything, static '
-              'values will be used.' % SUPPORTED_MIDDLEWARE.keys()))
+              'values will be used.' % providers.keys()))
 
-    for provider_name, provider in SUPPORTED_MIDDLEWARE.items():
+    for provider_name, provider in providers.items():
```

Comment:
should formatters bring options too?

Reply:
is any needed or foreseen?

Reply:
I guess not right now, but I would expect some flexibility here. 
We can always add it later, so resolving.